nixos/modules/services/libeufin/common.nix

```diff
@@ -0,0 +1,161 @@
+# TODO: create a common module generator for Taler and Libeufin?
+libeufinComponent:
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+{
+  options.services.libeufin.${libeufinComponent} = {
+    enable = lib.mkEnableOption "libeufin core banking system and web interface";
+    package = lib.mkPackageOption pkgs "libeufin" { };
+    debug = lib.mkEnableOption "debug logging";
+    # TODO: find a better way to
+    openFirewall = lib.mkEnableOption "Open ports in the firewall";
+    createLocalDatabase = lib.mkEnableOption "automatic creation of a local postgres database";
+  };
+
+  config =
+    let
+      cfgMain = config.services.libeufin;
+      cfg = cfgMain.${libeufinComponent};
+
+      serviceName = "libeufin-${libeufinComponent}";
+
+      isNexus = libeufinComponent == "nexus";
+
+      # get database name from config
+      # TODO: should this always be the same db? In which case, should this be an option directly under `services.libeufin`?
+      dbName =
+        lib.removePrefix "postgresql:///"
+          cfg.settings."libeufin-${libeufinComponent}db-postgres".CONFIG;
```

Comment:
I'll just leave a comment here, to mark it as a point of discussion. If `nexus` and `bank` always uses the same database url and credentials, should we configure it directly under `services.libeufin`?

Reply:
Yeah, I think this might be necessary, unless there is a way to give permissions across databases.